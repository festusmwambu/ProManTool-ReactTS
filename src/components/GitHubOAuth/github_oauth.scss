.oauth-github {
    width: 165px;
    height: 25px;
    background-color: #000000;
    color: white;
    display: flex;
    align-items: center;
    border-radius: 54px;
    cursor: pointer;
    margin: 10px 0;

    &:hover {
        background-color: #212121;
    }

    &__text {
        margin: 0;
        font-size: 13px;
    }

    &__icon {
        margin: 0 10px;
    }
}