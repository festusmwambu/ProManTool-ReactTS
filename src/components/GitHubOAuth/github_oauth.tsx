import { v4 as uuidv4 } from "uuid";
import qs from "qs";

import { trackEvent } from "../../utils/ga_util";
import { OAUTH_GITHUB } from "../../config";
import "./github_oauth.scss";



interface GitHubOAuthProps {
    onSuccess(code: string, state: string): void;
    text?: string;
};

const GitHubOAuth = ({ onSuccess, text = "Auth with Github" }: GitHubOAuthProps) => {
    const openModal = () => {
        trackEvent({
            category: "OAUTH",
            action: "Opened GitHub OAuth Modal",
        });
    };

    const state = uuidv4();

    const url = `https://github.com/login/oauth/authorize?scope=user:email&state=${state}&client_id=${OAUTH_GITHUB.CLIENT_ID}`;

    const popup = window.open(url, "", "width=400, height=600, left=400, top=50");

    popup?.addEventListener("beforeunload", (event) => {
        const { code } = qs.parse(popup?.location.search, { ignoreQueryPrefix: true });

        if (code) {
            onSuccess(code as string, state);
        }
    });

    return (
        <div className="oauth-github" onClick={openModal}>
            <i className="oauth-github__icon fa fa-github" aria-hidden="true" />
            <p className="oauth-github__text">{text}</p>
        </div>
    );
};

export default GitHubOAuth;