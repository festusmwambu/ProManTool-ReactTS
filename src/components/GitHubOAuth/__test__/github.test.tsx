import { render, screen } from '@testing-library/react';

import GitHubOAuth from '../github_oauth';



describe("<GitHubOAuth />", () => {
  it('Matches snapshot', () => {
    render(
      <GitHubOAuth onSuccess={jest.fn()} />
    );
    
    // Use screen.queryByTestId method
    const GitHubOAuthElement = screen.queryByTestId("gitHubOAuth-element");
        
    // Expectations
    expect(GitHubOAuthElement).toMatchSnapshot();
  });

  it('Receives text', () => {
    const testText = 'testText';

    render(
      <GitHubOAuth onSuccess={jest.fn()} text={testText} />
    );
    
    const GitHubOAuthElement = screen.queryByTestId("gitHubOAuth-element");
    
    expect(GitHubOAuthElement).toHaveTextContent('testText');
  });
});