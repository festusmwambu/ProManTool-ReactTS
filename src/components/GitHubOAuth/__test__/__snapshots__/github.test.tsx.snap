// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GitHubOAuth /> Matches snapshot 1`] = `
<div
  className="oauth-github"
>
  <i
    aria-hidden="true"
    className="oauth-github__icon fab fa-github"
  />
  <p
    className="oauth-github__text"
  >
    Auth with Github
  </p>
</div>
`;